_unittest/conftest.py

```diff
@@ -85,27 +85,37 @@ def my_setup(self):
         self.aedtapps = []
         self.edbapps = []
 
-    def my_teardown(self):
-        try:
-            oDesktop = sys.modules["__main__"].oDesktop
-        except Exception as e:
-            oDesktop = None
-        if oDesktop:
-            oDesktop.ClearMessages("", "", 3)
-        for edbapp in self.edbapps[::-1]:
-            try:
-                edbapp.close_edb()
-            except:
-                pass
-        del self.edbapps
-        for aedtapp in self.aedtapps[::-1]:
+    def my_teardown(self, close_desktop=False):
+        if close_desktop and not is_ironpython:
+            d = Desktop(desktop_version, non_graphical, False)
+            d.release_desktop()
+        else:
             try:
-                aedtapp.close_project(None, False)
-            except:
-                pass
-        del self.aedtapps
+                oDesktop = sys.modules["__main__"].oDesktop
+            except Exception as e:
+                oDesktop = None
+            if oDesktop:
+                oDesktop.ClearMessages("", "", 3)
+            for edbapp in self.edbapps[::-1]:
+                try:
+                    edbapp.close_edb()
+                except:
+                    pass
+            del self.edbapps
+            for aedtapp in self.aedtapps[::-1]:
+                try:
+                    aedtapp.close_project(None, False)
+                except:
+                    pass
+            del self.aedtapps
 
     def add_app(self, project_name=None, design_name=None, solution_type=None, application=None):
+        if "oDesktop" not in dir(sys.modules["__main__"]):
+            if self.aedtapps:
+                self.aedtapps[0].release_desktop()
```

Comment:
Is this a safety mechanism? 
I don't get when it will execute this release_desktop

Reply:
you are right. it should be in the teardown method.